admin/module/catalog-admin-module-sources/src/test/java/org/codice/ddf/catalog/admin/plugin/AdminSourcePollerTest.java

```diff
@@ -0,0 +1,129 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p/>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p/>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+
+package org.codice.ddf.catalog.admin.plugin;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyMap;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.util.ArrayList;
+import java.util.Dictionary;
+import java.util.HashMap;
+import java.util.Hashtable;
+import java.util.List;
+import java.util.Map;
+
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.osgi.framework.ServiceReference;
+import org.osgi.service.cm.Configuration;
+import org.osgi.service.cm.ConfigurationAdmin;
+
+import ddf.catalog.service.ConfiguredService;
+import ddf.catalog.source.Source;
+import ddf.catalog.source.opensearch.OpenSearchSource;
+
+public class AdminSourcePollerTest {
+
+    public static final String CONFIG_PID = "OpenSearchSourceFederated";
+
+    public static final String FPID = "OpenSearchSource";
+
+    public static MockedAdminSourcePoller poller;
+
+    @BeforeClass
+    public static void setup() {
+        poller = new AdminSourcePollerTest().new MockedAdminSourcePoller(null);
+    }
+
```

Comment:
Seems like there should be more tests.  What happens if service.isAvailable() throws an exception?  What happens if a serivce pid you are looking up does not have a configuration?